lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,472 @@
+            .iter()
+            .map(|path| {
+                if path == "." || path == "/" {
+                    return ("/".to_string(), Path::new("/").to_path_buf());
+                }
+
+                let key = format!("./{path}");
+                let value = Path::new(&format!("/{path}")).to_path_buf();
+
+                (key, value)
+            })
+            .collect();
+
+        Some(include)
+    }
+
+    fn get_command(
+        modules: &[wapm_toml::Module],
+        bin_or_lib: BinOrLib,
+    ) -> Option<Vec<wapm_toml::Command>> {
+        match bin_or_lib {
+            BinOrLib::Bin => Some(
+                modules
+                    .iter()
+                    .map(|m| {
+                        wapm_toml::Command::V1(wapm_toml::CommandV1 {
+                            name: m.name.clone(),
+                            module: m.name.clone(),
+                            main_args: None,
+                            package: None,
+                        })
+                    })
+                    .collect(),
+            ),
+            BinOrLib::Lib | BinOrLib::Empty => None,
+        }
+    }
+
+    /// Returns the dependencies based on the `--template` flag
+    fn get_dependencies(&self) -> Option<HashMap<String, String>> {
+        Some({
+            match self.template.as_deref() {
+                Some("js") => {
+                    let mut map = HashMap::default();
+                    map.insert("python".to_string(), "quickjs/quickjs@latest".to_string());
+                    map
+                }
+                Some("python") => {
+                    let mut map = HashMap::default();
+                    map.insert("python".to_string(), "python/python@latest".to_string());
+                    map
+                }
+                _ => HashMap::default(),
+            }
+        })
+    }
+
+    // Returns whether the template for the wapm.toml should be a binary, a library or an empty file
+    fn get_bin_or_lib(&self) -> Result<BinOrLib, anyhow::Error> {
+        match (self.empty, self.bin, self.lib) {
+            (true, true, _) | (true, _, true) => Err(anyhow::anyhow!(
+                "cannot combine --empty with --bin or --lib"
+            )),
+            (true, false, false) => Ok(BinOrLib::Empty),
+            (_, true, true) => Err(anyhow::anyhow!(
+                "cannot initialize a wapm manifest with both --bin and --lib, pick one"
+            )),
+            (false, true, _) => Ok(BinOrLib::Bin),
+            (false, _, true) => Ok(BinOrLib::Lib),
+            _ => Ok(BinOrLib::Bin),
+        }
+    }
+
+    fn get_bindings(target_file: &Path, bin_or_lib: BinOrLib) -> Option<wapm_toml::Bindings> {
+        match bin_or_lib {
+            BinOrLib::Bin | BinOrLib::Empty => None,
+            BinOrLib::Lib => target_file.parent().and_then(|parent| {
+                walkdir::WalkDir::new(parent)
+                    .min_depth(1)
+                    .max_depth(3)
```

Comment:
I don't know if we want to artificially limit the search depth to 1-3 levels, here.

Reply:
It's supposed to prevent accidentally searching things like `.git/objects/...` and build directories, which would maybe make it slow. `wasmer init` is not supposed to be perfect, it's just supposed to give a rough template that the user can edit.